### PR TITLE
Change the pages-improved stat to not count pages created

### PR DESCRIPTION
**Pull request by samwilson**
Bug: T206817

### CODE REVIEW

**Comment by samwilson** [comment]
> You're counting revisions that didn't create a page here. Let me know if I can help figuring this out.

I'd love you to take a look at it! My brain is full of mush at the moment, with a cold.

It's meant to be counting revisions that didn't create a page, isn't it? i.e. those are edits to existing pages. Compare to https://github.com/wikimedia/eventmetrics/blob/0a8a8f3d8d012d8ff160610b3985a221b20c0ca6/src/AppBundle/Repository/EventWikiRepository.php#L259

**Comment by samwilson** [comment]
> That begs the question: do we even need to fetch `edited` and `created` in EventRepository::getEditStats()? We can simply `count()` the page IDs.

Yeah, I was wondering about that. Would save the extra query. Is getEditStats needed for anything else?

**Comment by MusikAnimal** [comment]
> > That begs the question: do we even need to fetch `edited` and `created` in EventRepository::getEditStats()? We can simply `count()` the page IDs.
> 
> Yeah, I was wondering about that. Would save the extra query. Is getEditStats needed for anything else?

The number of edits, that's it I think.

**Comment by samwilson** [comment]
Take a look at this now, see what you think. I deleted `EventRepository::getEditStats()`.

File: src/AppBundle/Service/EventProcessor.php [hunk 1]
```diff
@@ -348,18 +348,19 @@ private function setContributionsWikipedias(EventWiki $wiki, EventWikiRepository
...
-        $ret = $this->eventRepo->getEditStats($dbName, $pageIds, $start, $end, $usernames);
-
         $diff = $ewRepo->getBytesChanged($this->event, $dbName, $pageIds, $usernames);
 
-        $this->pagesCreated += $ret['created'];
-        $this->pagesImproved += $ret['edited'];
-        $this->edits += $ret['edits'];
+        $totalCreated = count($pageIdsCreated);
+        $totalEdited = count($pageIdsEdited);
+        $total = count($pageIds);
```
Comment (MusikAnimal) [inline src/AppBundle/Service/EventProcessor.php]: What about `$totalCreated + $totalEdited`, then we can remove line 350?

File: src/AppBundle/Service/EventProcessor.php [hunk 2]
```diff
@@ -348,18 +348,19 @@ private function setContributionsWikipedias(EventWiki $wiki, EventWikiRepository
...
+        $this->pagesCreated += $totalCreated;
+        $this->pagesImproved += $totalEdited;
+        $this->edits += $total;
```
Comment (MusikAnimal) [inline src/AppBundle/Service/EventProcessor.php]: `pagesCreated` is the number of unique pages improved, `pagesCreated` is the number of unique pages created. Neither is a count of the edits made to those pages, right? I don't think we're querying for that anymore (it was one of the things `getEditStats()` did).

Comment (samwilson) [reply src/AppBundle/Service/EventProcessor.php]: doh! Yes, you're quite right. :) I'll fix it up.

File: src/AppBundle/Service/EventProcessor.php (hunk 1 above)
Comment (samwilson) [reply src/AppBundle/Service/EventProcessor.php]: I think we still need $pageIds to get the bytes changed. And I've just readded getEditStats too...

**Comment by samwilson** [comment]
> Did we want to make `getEditStats()` only count edits, and remove `COUNT(DISTINCT(rev_page)) AS edited` and `IFNULL(SUM(CASE WHEN rev_parent_id = 0 THEN 1 ELSE 0 END), 0) AS created`? The "edited" and "created" aren't being used anymore, so maybe even rework `getEditStats()` to be `getTotalEditCount()` and have it return an `int`. What do you think?

Good idea, I thought exactly the same thing last night after I turned the computer off. :)

Done now, see what you think.

**Pull request merged**